packages/buckets/src/api/index.ts

```diff
@@ -658,15 +775,65 @@ export async function bucketsPullPathAccessRoles(
   return typedRoles
 }
 
+/**
+ * @internal
+ */
+export async function bucketsDefaultArchiveConfig(api: GrpcConnection, key: string, ctx?: ContextInterface) {
+  logger.debug('default archive config request')
+  const req = new DefaultArchiveConfigRequest()
+  req.setKey(key)
+  const res: DefaultArchiveConfigResponse = await api.unary(APIService.DefaultArchiveConfig, req, ctx)
+  const config = res.getArchiveConfig()
+  if (!config) {
+    throw new Error('no archive config returned')
+  }
+  return fromProtoArchiveConfig(config)
+}
+
+/**
+ * @internal
+ */
+export async function bucketsSetDefaultArchiveConfig(
+  api: GrpcConnection,
+  key: string,
+  config: ArchiveConfig,
+  ctx?: ContextInterface,
+) {
+  logger.debug('set default archive config request')
+  const req = new SetDefaultArchiveConfigRequest()
+  req.setKey(key)
+  req.setArchiveConfig(toProtoArchiveConfig(config))
+  await api.unary(APIService.SetDefaultArchiveConfig, req, ctx)
+  return
+}
+/**
+ * An object to configure options for Archive.
+ */
+export interface ArchiveOptions {
+  /**
+   * Provide a custom ArchiveConfig to override use of the default.
+   */
+  archiveConfig?: ArchiveConfig
+}
+
 /**
  * archive creates a Filecoin bucket archive via Powergate.
  * @internal
  * @param key Unique (IPNS compatible) identifier key for a bucket.
+ * @param options Options that control the behavior of the bucket archive
  */
-export async function bucketsArchive(api: GrpcConnection, key: string, ctx?: ContextInterface) {
+export async function bucketsArchive(
+  api: GrpcConnection,
+  key: string,
+  options?: ArchiveOptions,
```

Comment:
This is a pretty important breaking change here, especially just because they both take objects as args. I think it most cases, we're not using ctx directly, so it should be pretty safe. But we'll have to be sure this works with slates usage.

Reply:
Yea, I don't even understand context. I never see it used anywhere within the sdk, but it looks to just be available internally. All public apis don't allow it to be passed in as far as I can tell.

Reply:
just a note. this shouldn't be a breaking change. we don't promote the use of these internal functions. on the public api, `Buckets.archive` options was just added as an optional second param, so non-breaking there too

Reply:
Great, thanks for the confirmation.